monin_obukhov/include/monin_obukhov.inc

```diff
@@ -16,190 +16,25 @@
-
-if(drag_min_moist.le.0.0)  call error_mesg( &
-        'MONIN_OBUKHOV_INIT in MONIN_OBUKHOV_MOD', &
-        'drag_min_moist in monin_obukhov_mod must be >= 0.0', FATAL)
-
-if(drag_min_mom.le.0.0)  call error_mesg( &
-        'MONIN_OBUKHOV_INIT in MONIN_OBUKHOV_MOD', &
-        'drag_min_mom in monin_obukhov_mod must be >= 0.0', FATAL)
-
-if(stable_option < 1 .or. stable_option > 2) call error_mesg( &
-        'MONIN_OBUKHOV_INIT in MONIN_OBUKHOV_MOD', &
-        'the only allowable values of stable_option are 1 and 2', FATAL)
-
-if(stable_option == 2 .and. zeta_trans < 0) call error_mesg( &
-        'MONIN_OBUKHOV_INIT in MONIN_OBUKHOV_MOD', &
-        'zeta_trans must be positive', FATAL)
-
-b_stab = 1.0/rich_crit
-r_crit = 0.95*rich_crit  ! convergence can get slow if one is
-                         ! close to rich_crit
-
-sqrt_drag_min_heat = 0.0
-if(drag_min_heat.ne.0.0) sqrt_drag_min_heat = sqrt(drag_min_heat)
-
-sqrt_drag_min_moist = 0.0
-if(drag_min_moist.ne.0.0) sqrt_drag_min_moist = sqrt(drag_min_moist)
-
-sqrt_drag_min_mom = 0.0
-if(drag_min_mom.ne.0.0) sqrt_drag_min_mom = sqrt(drag_min_mom)
-
-lambda     = 1.0 + (5.0 - b_stab)*zeta_trans   ! used only if stable_option = 2
-rich_trans = zeta_trans/(1.0 + 5.0*zeta_trans) ! used only if stable_option = 2
-
-module_is_initialized = .true.
-
-return
-end subroutine monin_obukhov_init
-
-!=======================================================================
-
-subroutine monin_obukhov_end
-
-module_is_initialized = .false.
-
-end subroutine monin_obukhov_end
-
-!=======================================================================
-
-subroutine mo_drag_1d &
+subroutine MO_DRAG_1D_ &
          (pt, pt0, z, z0, zt, zq, speed, drag_m, drag_t, drag_q, &
           u_star, b_star, avail)
 
-real, intent(in)   , dimension(:) :: pt, pt0, z, z0, zt, zq, speed
-real, intent(inout), dimension(:) :: drag_m, drag_t, drag_q, u_star, b_star
-logical, intent(in), optional, dimension(:) :: avail
+real(kind=FMS_MO_KIND_), intent(in)   , dimension(:) :: pt, pt0, z, z0, zt, zq, speed
+real(kind=FMS_MO_KIND_), intent(inout), dimension(:) :: drag_m, drag_t, drag_q, u_star, b_star
+logical, intent(in), optional, dimension(:)          :: avail
 
-logical            :: lavail, avail_dummy(1)
-integer            :: n, ier
+logical                                              :: lavail, avail_dummy(1)
+integer                                              :: n, ier
 
-integer, parameter :: max_iter = 20
-real   , parameter :: error=1.e-04, zeta_min=1.e-06, small=1.e-04
+integer, parameter                                   :: max_iter = 20
+integer, parameter                                   :: lkind    = FMS_MO_KIND_
+real(kind=FMS_MO_KIND_), parameter                   :: error    = 1.0E-04_lkind, &
+                                                        zeta_min = 1.0E-06_lkind, &
+                                                        small    = 1.0E-04_lkind
 
-! #include "monin_obukhov_interfaces.h"
```

Comment:
should we keep this commented out code?

Reply:
I don't recall if I was told to remove this or why I did this so I will follow your lead on whether or not to keep it in the code

Reply:
Let's keep it in and let the future people remove it if they want to.

Reply:
yes ma'am